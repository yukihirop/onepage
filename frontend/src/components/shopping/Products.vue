<template lang="pug">
  div
    h1.title All Products
      p {{ length }} products
      table.table.is-striped
        thead
          tr
            th Name
            th Description
            th Price
        tbody
          tr(v-for='product in products' track-by='id')
            td {{ product.name }}
            td {{ product.description }}
            td ${{ product.price }}
            td
              button.button.is-info(@click='addToCart(product)') Add to cart
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
const namespace = 'shoppingModule'

export default {
  name: 'app',
  computed: mapGetters(namespace, {
    products: 'allProducts',
    length: 'getNumberOfProducts'
  }),
  methods: mapActions(namespace, [
    'addToCart'
  ])
}
</script>