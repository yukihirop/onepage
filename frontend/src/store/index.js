import Vue from 'vue'
import Vuex from 'vuex'
import shoppingModule from './modules/shopping'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

export default new Vuex.Store({
  strict: debug,
  modules: {
    shoppingModule
  }
})