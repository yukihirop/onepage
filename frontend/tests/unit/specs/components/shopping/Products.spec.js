import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import Products from '@/components/shopping/Products.vue'
import store from '@/store'
import factoryProducts from '@specs/factories/components/shopping/Products'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Products.vue', () => {
  it('初期状態のレンダリングのテスト', () => {
    const wrapper = shallow(Products, { store, localVue })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('storeを用意してレンダリングのテスト', () => {
    const mockedStore = {
      modules: {
        shoppingModule: {
          namespaced: true,
          getters: {
            allProducts: jest.fn().mockReturnValue(factoryProducts.allProducts),
            getNumberOfProducts: jest.fn().mockReturnValue(3)
          },
          methods: jest.fn()
        }
      }
    }
    const store = new Vuex.Store(mockedStore)
    const wrapper = shallow(Products, { store, localVue })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
