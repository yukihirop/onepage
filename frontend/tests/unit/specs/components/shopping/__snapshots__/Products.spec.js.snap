// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products.vue storeを用意してレンダリングのテスト 1`] = `
<div>
    <h1 class="title">All Products
        <p>3 products</p>
        <table class="table is-striped">
            <thead>
                <tr>
                    <th>Name</th>
                    <th>Description</th>
                    <th>Price</th>
                </tr>
            </thead>
            <tbody>
                <tr track-by="id">
                    <td>a</td>
                    <td>A</td>
                    <td>$1</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
                <tr track-by="id">
                    <td>b</td>
                    <td>B</td>
                    <td>$2</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
                <tr track-by="id">
                    <td>c</td>
                    <td>C</td>
                    <td>$3</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
            </tbody>
        </table>
    </h1>
</div>
`;

exports[`Products.vue 初期状態のレンダリングのテスト 1`] = `
<div>
    <h1 class="title">All Products
        <p>3 products</p>
        <table class="table is-striped">
            <thead>
                <tr>
                    <th>Name</th>
                    <th>Description</th>
                    <th>Price</th>
                </tr>
            </thead>
            <tbody>
                <tr track-by="id">
                    <td>COBOL 101 vintage</td>
                    <td>Learn COBOL with this vintage programming book</td>
                    <td>$399</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
                <tr track-by="id">
                    <td>Sharp C2719 curved TV</td>
                    <td>Watch TV like never before with the brand new curved screen technology</td>
                    <td>$1995</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
                <tr track-by="id">
                    <td>Remmington X mechanical keyboard</td>
                    <td>Excellent for gaming and typing, this Remmington X keyboard features tactile, clicky switches for speed and accuracy</td>
                    <td>$595</td>
                    <td>
                        <button class="button is-info">Add to cart</button>
                    </td>
                </tr>
            </tbody>
        </table>
    </h1>
</div>
`;
